compiler/back_end/cpp/generated_code_templates

```diff
@@ -391,11 +392,14 @@ ${write_fields}
 // ** ok_method_test ** ////////////////////////////////////////////////////////
     // If we don't have enough information to determine whether ${field} is
     // present in the structure, then structure.Ok() should be false.
-    if (!has_${field}.Known()) return false;
-    // If ${field} is present, but not Ok(), then structure.Ok() should be
-    // false.  If ${field} is not present, it does not matter whether it is
-    // Ok().
-    if (has_${field}.ValueOrDefault() && !${field}.Ok()) return false;
+    {
+      const auto emboss_reserved_local_field_present = ${existence_condition};
```

Comment:
So I *think* this is intended to inline the contents of `has_<field>`, is that correct? And the goal of that is to reuse a cached instance of a field that is referenced in multiple `has_<field>` methods? Looking at the generated code now it's...pretty verbose. I'm actually a little surprised this would reduce code size.

Reply:
Check out https://github.com/google/emboss/pull/235 which I think is more correct as to what Ben suggested I implement and the code is much cleaner.  Some compiler specific attributes could be elided, let me know.